lighthouse-core/test/lib/dependency-graph/simulator/network-analyzer-test.js

```diff
@@ -385,4 +388,30 @@ describe('DependencyGraph/Simulator/NetworkAnalyzer', () => {
       assert.equal(mainDocument.url, 'https://www.example.com');
     });
   });
+
+  describe('#resolveRedirects', () => {
+    it('should resolve to the same document when no redirect', async () => {
+      const records = await NetworkRecords.request(devtoolsLog, {computedCache: new Map()});
+
+      const mainDocument = NetworkAnalyzer.findMainDocument(records, 'https://pwa.rocks/');
+      const finalDocument = NetworkAnalyzer.resolveRedirects(mainDocument);
+      assert.equal(mainDocument.url, finalDocument.url);
+      assert.equal(finalDocument.url, 'https://pwa.rocks/');
+    });
+
+    it('should resolve to the final document with redirects', async () => {
+      const records = await NetworkRecords.request(devtoolsLogWithRedirect, {
+        computedCache: new Map(),
+      });
+
+      const mainDocument = NetworkAnalyzer.findMainDocument(records, 'http://www.vkontakte.ru/');
+      const finalDocument = NetworkAnalyzer.resolveRedirects(mainDocument);
+      assert.notEqual(mainDocument.url, finalDocument.url);
+      assert.equal(finalDocument.url, 'https://m.vk.com/');
+    });
+
+    it('should throw if not given a request', () => {
```

Comment:
Since the parameter is typechecked to not allow undefined, we can consider this case covered trivially by the typesystem, so let's delete this test case.

Reply:
Done!